src/nccl_ofi_msgbuff.c

```diff
@@ -41,12 +46,28 @@ nccl_ofi_msgbuff_t *nccl_ofi_msgbuff_init(uint16_t buffer_size)
 
 error:
 	if (msgbuff) {
-		if (msgbuff->buff) free(msgbuff->buff);
+		if (msgbuff->buff) {
+			free(msgbuff->buff);
+		}
 		free(msgbuff);
 	}
 	return NULL;
 }
 
+static inline void increase_with_wraparound(const nccl_ofi_msgbuff_t *msgbuff, uint16_t *ctr)
+{
+	if (OFI_UNLIKELY(*ctr == (msgbuff->size - 1))) {
```

Comment:
This seems overly complicated.  Why are we not just forcing size to be a power of two and doing masked math like we used to?